File: server/contract-review-copilot/backend/src/password_policy.py
from __future__ import annotations

import re


PASSWORD_POLICY_MESSAGE = "密码必须至少 8 位，并包含大写字母、小写字母和数字"

_PASSWORD_POLICY_PATTERN = re.compile(r"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$")


def get_password_validation_error(password: str) -> str | None:
    normalized_password = password.strip()
    if not normalized_password:
        return "密码不能为空"
    if not _PASSWORD_POLICY_PATTERN.match(normalized_password):
        return PASSWORD_POLICY_MESSAGE
    return None

File: server/contract-review-copilot/frontend/src/lib/passwordPolicy.ts
export const PASSWORD_POLICY_MESSAGE = '密码必须至少 8 位，并包含大写字母、小写字母和数字'

const PASSWORD_POLICY_PATTERN = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$/

export function getPasswordValidationError(password: string): string | null {
  const normalizedPassword = password.trim()
  if (!normalizedPassword) {
    return '请输入密码'
  }
  if (!PASSWORD_POLICY_PATTERN.test(normalizedPassword)) {
    return PASSWORD_POLICY_MESSAGE
  }
  return null
}
